bitem::parse: fixup uncaught exception.

ttys3 opened the pull request:
std::stoull may throw an exception cause php terminated.
    for example:
```php
    bitem::parse('d5:infod6:lengthi364e4:name8:test.rare3:inti1548400939e4:listl5:UTF-83:GBKe6:string13:uTorrent/3130e');
```
    exception:
```
    terminate called after throwing an instance of 'std::invalid_argument'
```

Frederick888 commented:
Thanks for the PR!

Hmmm... I think `bitem::load` needs the same fix and perhaps it makes sense to catch `std::out_of_range` as well. Would you like to make the changes or you want me to do this afterwards?

Commit:
fixup uncaught C++ exception.
std::string::substr may cause std::out_of_range exception
std::stoull and std::stoll may cause std::invalid_argument exception
both will cause php terminated.

related method:
bdict::parse
bint::parse
bstr::parse
bitem::escape_key  -> used by bdict::search
bitem::get_current_key  -> used by bdict/blist  del_path get_path set_path

ttys3 commented:
I've made a macro wrap named `CALL_AND_HANDLE` to handle the exception.

`std::string::substr` may cause std::out_of_range exception
`std::stoull` and `std::stoll` may cause std::invalid_argument exception
both will cause php terminated.

related method:
bdict::parse 
bint::parse
bstr::parse
bitem::escape_key  -> used by bdict::search
bitem::parse
bitem::load

maybe related:
bitem::get_current_key  -> used by bdict/blist  del_path get_path set_path

Frederick888 commented:
The macro looks good to me :) I'll publish a new release asap.
